### ISSUE TITLE
Create and publish a V0.0.1 document release

### ISSUE THREAD

**Issue by axel-h** [riscv-non-isa/riscv-asm-manual#115]
Now that https://github.com/riscv-non-isa/riscv-asm-manual/issues/112 has been merged, there should be a first release of the document with the goal to have this also on the RISC-V docs website. Could still be marked as draft or prerelease.

**Comment by wmat** [comment]
Is this manual an artificate under the toolchain TG?

**Comment by apazos** [comment]
@axel-h , we discussed your CI workflow request at the last Toolchain SIG meeting on 11/21 and we agreed on adding a CI workflow, but not an official release yet.

In addition, the spec versioning scheme will change with the new BoD Policy, we should wait until it is settled.

Can you explain your urgency to have a first release?

Adding @cmuellner to comment as well.

**Comment by axel-h** [comment]
There is no urgency, it would just be nice to have something one can refer to in other document - besides just a commit ID in a repository.

**Comment by axel-h** [comment]
It seems merging the workflow was not enough, this also needs a workflow approval somewhere. The CI job
https://github.com/riscv-non-isa/riscv-asm-manual/actions/runs/12198852957/job/34031552715 tried to upload the artifacts after a push to `main` to https://github.com/riscv-non-isa/riscv-asm-manual/releases/tag/untagged-b05b87e58d03d44b1e68, but this gives a 404  and the draft documents for latest `main`do not show up under https://github.com/riscv-non-isa/riscv-asm-manual/releases. It workes in my fork at https://github.com/axel-h/riscv-asm-manual/releases when testing this. Could you have a look at "Settings" -> "Actions" -> ""General" -> "Workflow permissions", that the `GITHUB_TOKEN` has read and write access. Default might be just read access.

**Comment by wmat** [comment]
The permissions for the Action are read/write.  The reason the PDF doesn't
show up in the releases section on GitHub is because it only uploads as an
asset of the Action, not a release.  If you would like it to create a draft
release on each merge, you need another action setup such as is found in
the riscv-isa-manual repo here
https://github.com/riscv/riscv-isa-manual/actions/workflows/merge-and-release.yml,
if that is something you'd like on this repository you can add the Action
or let me know and I can do it.  Optionally, you can create a release
manually.

**Comment by cmuellner** [comment]
I really appreciate the attempts to get CI/CD up and running.

However, I think we should copy from the psABI repo, as the workflow there has shown to work quite well.
We might make little adjustments (where necessary) or use more recent versions of external jobs, but overall, we should just do what can be found there: https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/master/.github/workflows/build.yml

@axel-h, if you are interested then feel free to adjust this repo's workflow accordingly. We may also look at https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/master/.github/workflows/remove-old-artifacts.yml.

For the particular issue that is discussed here, the psABI workflow file does what is expected by the "Upload release asset" step of the "draft-release" step. See https://github.com/riscv-non-isa/riscv-elf-psabi-doc/blob/master/.github/workflows/build.yml#L59.

**Comment by cmuellner** [comment]
@axel-h, I've created a PR to add the CI/CD scripts from the docs-template repo (https://github.com/riscv/docs-spec-template/tree/main/.github). See #118.

To avoid fragmentation, I also filed similar PRs to harmonize the CI/CD scripts of the C-API and the Toolchain Conventions as well. See https://github.com/riscv-non-isa/riscv-toolchain-conventions/pull/66 and https://github.com/riscv-non-isa/riscv-c-api-doc/pull/98.

**Comment by wmat** [comment]
Just wondering if someone could go ahead and create this release soon?  I get asked almost weekly where the latest release is.  I'm happy to help do this if everyone is busy.  Just ping me.

**Comment by apazos** [comment]
Hi @wmat what do you plan to improve from the draft already available at https://github.com/riscv-non-isa/riscv-asm-manual/releases?

**Comment by wmat** [comment]
> Hi [@wmat](https://github.com/wmat) what do you plan to improve from the draft already available at https://github.com/riscv-non-isa/riscv-asm-manual/releases?

All I was looking for is the addition of the "create a release" step within the GH action in the same way it occurs in other repositories.  That way the top of tree release will always have a link from the repo's landing page on the right hand side.  The reason for this is that there is a broken link in the Unprivileged ISA manual (in Chapter 2) to the ASM manual and the person who pointed it out is asking for a link to fix it with.

**Comment by apazos** [comment]
Thanks @wmat for clarifying what you want to fix, please go ahead with the change. 
Please add the missing GH action.
I will then try to propagate your change to our other doc repos to keep them in sync with GH workflows.

**Comment by wmat** [comment]
It looks like nothing is missing, however the 'Create Release' portion of the GH action requires that it be manually dispatched.  Can I simply do that and create a release?

**Comment by apazos** [comment]
Yes, I think it is fine to proceed with this action, as it will generate a link to it on the repo's page, which can then be referenced in other documents. 

Please explain the process you do so I can replicate it in our other repos we own.

**Comment by cmuellner** [comment]
I don't fully get this. Do we really have a fully automated build pipeline and need to do pre-releases manually?

**Comment by axel-h** [comment]
My plan was to get something working that create a draft for every commit and makes this available. Just releases have to created manually. However, this does not full work at the moment, as the drafts don't show up.  Unfortunately, I did not find the time to update this.
